File: src/plugin/combat/magic/modern/teleport/LumbridgeTeleportSpell.java
package plugin.combat.magic.modern.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class LumbridgeTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 31;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(EARTH_RUNE, 1, AIR_RUNE, 3, LAW_RUNE, 1);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(3222, 3218, 0);
	}
	
	@Override
	public int spellId() {
		return 43;
	}
	
	@Override
	public double exp() {
		return 41;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.REGULAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/modern/teleport/LumbridgeTeleportSpell.kt
package plugin.interaction.combat.magic.modern.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/27/2017
 */
class LumbridgeTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 31
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.EARTH_RUNE, 1, MagicConstants.AIR_RUNE, 3, MagicConstants.LAW_RUNE, 1)
    }

    override fun destination(): WorldTile {
        return WorldTile(3222, 3218, 0)
    }

    override fun spellId(): Int {
        return 43
    }

    override fun exp(): Double {
        return 41.0
    }

    override fun book(): MagicBook {
        return MagicBook.REGULAR
    }
}